### PR TITLE
Messaging token refresh delegate should be able to return null token

### LINKED ISSUE #5355: Token in messaging:didReceiveRegistrationToken: should be nullable
#5339 fixes an issue when token is deleted, the `MessagingDelegate messaging:didReceiveRegistrationToken:` should be triggered with a nil token returned. Hence the token can be nullable. 
This is a breaking API that should be done in the next Firebase breaking change release.

### PR DESCRIPTION
**Pull request by chliangGoogle**
This fixes #5355

### CODE REVIEW

File: FirebaseMessaging/CHANGELOG.md
Comment (paulb777) [inline FirebaseMessaging/CHANGELOG.md:3]: Prefer issue number over PR number when possible for changelogs.

**Review by paulb777 — APPROVED** [review]
LGTM on green

File: FirebaseMessaging/Sources/Public/FirebaseMessaging/FIRMessaging.h
Comment (morganchen12) [inline FirebaseMessaging/Sources/Public/FirebaseMessaging/FIRMessaging.h:149]: nit: Use `nullable NSString *` (newer, more concise) over `NSString *__nullable`.

**Review by morganchen12 — APPROVED** [review]
LGTM, but please address comments before merging

**Comment by chliangGoogle** [comment]
@morganchen12 This breaks quickstart, but I don't see the quickstart-ios in this repo.

**Comment by paulb777** [comment]
@ryanwilson is exploring a firebase 7 quickstart branch for a similar CI issue with #6645

**Comment by chliangGoogle** [comment]
quickstart PR: https://github.com/firebase/quickstart-ios/pull/1056

File: FirebaseMessaging/CHANGELOG.md
Comment (chliangGoogle) [reply FirebaseMessaging/CHANGELOG.md]: Done.

File: FirebaseMessaging/Sources/Public/FirebaseMessaging/FIRMessaging.h
Comment (chliangGoogle) [reply FirebaseMessaging/Sources/Public/FirebaseMessaging/FIRMessaging.h:149]: @ryanwilson Need your approval since this is a public API change

**Comment by paulb777** [comment]
It looks like master is depending on this merging now? I'm seeing this failure - https://github.com/firebase/firebase-ios-sdk/runs/1211461516?check_suite_focus=true

**Comment by chliangGoogle** [comment]
I think I need to make more change in quickstart to make this PR green.

**Comment by chliangGoogle** [comment]
https://github.com/firebase/quickstart-ios/pull/1057 should fix the rest of brakage.

File: FirebaseMessaging/Sources/Public/FirebaseMessaging/FIRMessaging.h
Comment (ryanwilson) [reply FirebaseMessaging/Sources/Public/FirebaseMessaging/FIRMessaging.h:149]: +1 - Morgan's change makes sense to me and this LGTM

**Review by ryanwilson — APPROVED** [review]

File: FirebaseMessaging/Sources/Public/FirebaseMessaging/FIRMessaging.h
Comment (chliangGoogle) [reply FirebaseMessaging/Sources/Public/FirebaseMessaging/FIRMessaging.h]: Done!

**Pull request merged**
